Source/WebCore/crypto/mac/CryptoAlgorithmAES_GCMMac.cpp

```diff
@@ -43,12 +44,20 @@ ALLOW_DEPRECATED_DECLARATIONS_BEGIN
 ALLOW_DEPRECATED_DECLARATIONS_END
     if (status)
         return Exception { ExceptionCode::OperationError };
-
     memcpy(cipherText.data() + plainText.size(), tag.data(), desiredTagLengthInBytes);
 
     return WTFMove(cipherText);
 }
 
+static ExceptionOr<Vector<uint8_t>> encryptCryptoKitAES_GCM(const Vector<uint8_t>& iv, const Vector<uint8_t>& key, const Vector<uint8_t>& plainText, const Vector<uint8_t>& additionalData, size_t desiredTagLengthInBytes)
+{
+    Vector<uint8_t> cipherText(plainText.size() + 16);
+    if (CkAesGcm::encrypt(key.data(), key.size(), iv.data(), iv.size(), additionalData.data(), additionalData.size(), plainText.data(), plainText.size(), cipherText.data()))
+        return Exception { ExceptionCode::OperationError };
+    cipherText.resize(cipherText.size() - (16 - desiredTagLengthInBytes));
```

Comment:
What if desiredTagLengthInBytes is more than 16?

Reply:
This actually is checked one layer above. But I have added another check here regardless. Have also unified this 16 value in a constant and used that everywhere.